pkg/apis/wsecurity/v1/README.md

```diff
@@ -0,0 +1,47 @@
+# Security Data Package
+
+This package serves as the beating heart of Guard.
+
+It defines data structures that meet the [v1 interface](v1.go).
+
+## The wsecurity/v1 API
+
+The [v1 interface](v1.go) includes three interfaces:
+
+- Profile: describing a sample of a data type
+- Pile: accumulating multiple samples of a data type to enable learning.
+- Config: the rules describing what is expected from the data type.
+
+## Core Activity
+
+Per Sample:
+
+1. Profile.Profile(...sample...) - Create a profile from the sample
+1. Config.Decide(profile)  - Decide if it conforms to the config rules
+1. Pile.Add(profile)  - Add it to a pile.
+
+Periodically:
+
+1. Pile.Merge(someOtherPile) - Merge someOtherPile to Pile. Note that Piles consume other piles - once someOtherPile is added to pile it is no longer usable.
```

Comment:
Is the intent that the Pile is _empty_ or _unusable_?

If the latter, is there a particular reason to prevent the reuse of `Pile`s?

Reply:
The intent of the `no longer usable` is to free us to use internal data constructs of the other Pile if and when we choose to do so implementation-wise (to reduce overhead). 

The expected behavior is for piles to accumulate in QP instances  - then be delivered to a remote central learner service which will merge Piles from different  QP instances. once they are merged, these piles are dumped anyhow. I.e. the code looks like that:
```golang
... 
// get a pile sent from a QP instance
err := json.NewDecoder(req.Body).Decode(&pile)
...
// add the pile to a bigger pile we keep for all instances that serve the Knative service XYZ
record.pile.Merge(&pile) // pile no longer used
...
```